### PR TITLE
feat: support KnativeIngress

### PR DESCRIPTION
**Pull request by vikizhao156**
### Ⅰ. Describe what this PR did
support KnativeIngress

### CODE REVIEW

File: pkg/ingress/config/kingress_config.go [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+		XDSUpdater:               XDSUpdater,
+		annotationHandler:        annotations.NewAnnotationHandlerManager(),
+		clusterId:                clusterId,
+		globalGatewayName: namespace + "/" +
+			common.CreateConvertedName(clusterId, "global"),
+		watchedSecretSet:    sets.NewSet(),
+		namespace:           namespace,
+		mcpbridgeReconciled: true,
+		wasmPlugins:         make(map[string]*extensions.WasmPlugin),
+		http2rpcs:           make(map[string]*higressv1.Http2Rpc),
```
Comment (johnlanni) [inline pkg/ingress/config/kingress_config.go:106]: These fields seems no usage.

File: pkg/ingress/config/kingress_config.go [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+	IngressLog.Infof("resource type %s, configs number %d", typ, len(wrapperConfigs))
+	switch typ {
+	case gvk.Gateway:
+		return m.convertGateways(wrapperConfigs), nil
+	case gvk.VirtualService:
+		return m.convertVirtualService(wrapperConfigs), nil
+	case gvk.ServiceEntry:
+		return m.convertServiceEntry(wrapperConfigs), nil
+	case gvk.WasmPlugin:
+		return m.convertWasmPlugin(wrapperConfigs), nil
```
Comment (johnlanni) [inline pkg/ingress/config/kingress_config.go:190]: Seems no usage

File: pkg/ingress/kube/kingress/status.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+// newStatusSyncer creates a new instance
+func newStatusSyncer(localKubeClient, client kube.Client, controller *controller, namespace string) *statusSyncer {
+	return &statusSyncer{
+		client:           client.KIngress(),
+		controller:       controller,
+		watchedNamespace: namespace,
+		ingressLister:    client.KIngressInformer().Networking().V1alpha1().Ingresses().Lister(),
+		// search service in the mse vpc
```
Comment (johnlanni) [inline pkg/ingress/kube/kingress/status.go:52]:
```suggestion
```

**Review by johnlanni — CHANGES_REQUESTED** [review]

**Comment by johnlanni** [comment]
cc @SpecialYang

**Comment by vikizhao156** [comment]
Unused and duplicate code removed

**Pull request merged**